cypress/support/step_definitions/getterSetterCalls.js

```diff
@@ -16,25 +16,509 @@
+    } else {
+      // Reading fireboltCall object from the environment variable
+      fireboltCallObject = UTILS.getEnvVariable(CONSTANTS.RUNTIME).fireboltCall;
+      fireboltCallObjectErrorMessage =
+        'Unable to find the firebolt object in the runtime environment variable';
+    }
+
+    cy.then(() => {
+      // Failing the test when fireboltCall object not there
+      if (!fireboltCallObject) {
+        fireLog.assert(false, fireboltCallObjectErrorMessage);
+      } else {
+        let method = UTILS.resolveRecursiveValues(fireboltCallObject.method);
+        const param = UTILS.resolveRecursiveValues(fireboltCallObject.params);
+        const context = {};
+        const expected = CONSTANTS.RESULT;
+        appId =
+          appId === CONSTANTS.THIRD_PARTY_APP
+            ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
+            : appId === CONSTANTS.FIRST_PARTY_APP
+              ? UTILS.getEnvVariable(CONSTANTS.FIRST_PARTY_APPID)
+              : appId;
+        let action = CONSTANTS.ACTION_CORE.toLowerCase();
+
+        // Splitting the method name if it contains an underscore and using the first part to determine the action that decides sdk.
+        if (method && method.includes('_')) {
+          action = method.split('_')[0];
+          method = method.split('_')[1];
+        }
+
+        const additionalParams = {
+          method: method,
+          params: param,
+          context: context,
+          action: action,
+          expected: expected,
+          appId: appId,
+        };
+
+        if (appId == UTILS.getEnvVariable(CONSTANTS.FIRST_PARTY_APPID)) {
+          fireLog.info(
+            'Call from 1st party App, method: ' + method + ' params: ' + JSON.stringify(param)
+          );
+          cy.sendMessageToPlatformOrApp(CONSTANTS.PLATFORM, additionalParams);
+        } else {
+          if (
+            Cypress.env(CONSTANTS.TEST_TYPE) &&
+            Cypress.env(CONSTANTS.TEST_TYPE).toLowerCase() == CONSTANTS.MODULE_NAMES.LIFECYCLE
+          ) {
+            cy.fetchLifecycleHistory(appId);
+          }
+          fireLog.info(`Call from ${appId}, method: ${method} params: ${JSON.stringify(param)}`);
+          cy.sendMessageToPlatformOrApp(CONSTANTS.APP, additionalParams);
+        }
+      }
+    });
+  } else {
+    fireLog.assert(false, `${sdk} SDK not Supported`);
+  }
 });
+
+/**
+ * @module getterSetterCalls
+ * @function Given '(.+)' registers for the '(.*?)'(?: '(.*?)')? event
+ * @description Sending a message to platform or app to register an event
+ * @param {String} appId - app identifier.
+ * @param {String} sdk - sdk name.
+ * @param {String} fireboltCallKey - key name passed to look for firebolt call object in fireboltCallData.
+ * @example
+ * And '1st party app' registers for the 'Firebolt' 'CLOSEDCAPTION_SETTINGS' event
+ * And '3rd party app' registers for the 'Firebolt' 'CLOSEDCAPTION_SETTINGS' event
+ * And '1st party app' registers for the 'Firebolt' event
+ */
+Given(
```

Comment:
I think this one shares a good amount of logic as well, ideally we could create a helper function or two if needed to so that they all use the same logic?

Reply:
In upcoming enhancement ticket observed we are going to modify the glue and removing some fields, After removing some of the lines we need to remove. Is it to handle thins in new ticket?